source/val/validate_type.cpp

```diff
@@ -296,10 +291,25 @@ spv_result_t ValidateTypeFunction(ValidationState_t& _,
   return SPV_SUCCESS;
 }
 
+spv_result_t ValidateTypeForwardPointer(ValidationState_t& _,
```

Comment:
I would also suggest validating that the storage classes match if the first check passes.